### PR TITLE
WIP: remove nodes from the cluster

### PR DESCRIPTION
**Pull request by mssola**
Signed-off-by: Miquel Sabaté Solà <msabate@suse.com>

### CODE REVIEW

**Comment by mssola** [comment]
https://trello.com/c/uO999RGs/232-5-node-removal-salt-tasks

File: config/routes.rb [hunk 1]
```diff
@@ -5,6 +5,7 @@
 
   resource :dashboard, only: [:index]
   resource :updates, only: [:create]
+  resources :nodes, only: [:destroy]
```
Comment (MaximilianMeister) [inline config/routes.rb]: with having a `NodesController` do you think we can also add the bootstrap api to its responsibility? so `destroy` is a member method and `bootstrap` is a collection method

Comment (mssola) [inline config/routes.rb]: I don't know. This is a matter for the future I'd say. The dashboard and the setup controllers do not really follow any best practices, and we simply dumped methods into those. After the first release we'll have to re-organize things a bit.

For this PR, having this functionality here makes sense (and the "adding nodes" feature could re-use it but then again, it's a matter for re-organization in the future 😉).

Comment (vitoravelino) [inline config/routes.rb]: I totally agree on moving stuff to `NodesController` and being more resources oriented but only after the release. We'll have a sprint to cleanup stuff after the release and I think we should also prioritize this.

In #173 I added the actions to the dashboard controller but it should definitely be in `NodesController`. =/

**Comment by PI-Victor** [comment]
hmm, what's the current status of this?

**Comment by ereslibre** [comment]
@PI-Victor It hasn't got priority since this isn't required for 1.0, so I guess it's a post-1.0 consideration :)

**Comment by mssola** [comment]
Closing since I didn't work on this for a long time...

**Pull request closed without merging**